[
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "Hiii,\nHow do we make sure whether Attribute certificate authority is running properly or not?\nWhat kind of logging we get on terminal after enabling this?\nIn my docker-compose.yml file, I did - MEMBERSRVC_CA_ACA_ENABLED=true",
        "ts": "1473147633.000007"
    },
    {
        "user": "U0NK7T8SH",
        "text": "<@U0NK7T8SH|bryan-huang> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1473159137.000008"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Good morning <@U1RLZ5729> !",
        "ts": "1473174605.000009"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "If you set your log level to `info`, you will get a lot of info printed out to the log... I have added it as part of #2130 back in the day.",
        "ts": "1473174643.000010"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "See: <https:\/\/github.com\/hyperledger-archives\/fabric\/pull\/2130\/commits\/5a6b20179f52d4c2a3b07f46ebbd666b7e0ea1fe>",
        "attachments": [
            {
                "service_name": "GitHub",
                "title": "Graceful Membership Services shutdown: Introducing Stop() by JonathanLevi \u00b7 Pull Request #2130 \u00b7 hyperledger-archives\/fabric \u00b7 GitHub",
                "title_link": "https:\/\/github.com\/hyperledger-archives\/fabric\/pull\/2130\/commits\/5a6b20179f52d4c2a3b07f46ebbd666b7e0ea1fe",
                "text": "Description Added a Stop() at each *CA level (from Close()) and allowed the call to propagate down to the gRPC server. Now each ACA, ECA, TCA, TSLCA... each has a Stop() to match each Start(). ...",
                "fallback": "GitHub: Graceful Membership Services shutdown: Introducing Stop() by JonathanLevi \u00b7 Pull Request #2130 \u00b7 hyperledger-archives\/fabric",
                "thumb_url": "https:\/\/avatars0.githubusercontent.com\/u\/1161321?v=3&s=400",
                "from_url": "https:\/\/github.com\/hyperledger-archives\/fabric\/pull\/2130\/commits\/5a6b20179f52d4c2a3b07f46ebbd666b7e0ea1fe",
                "thumb_width": 420,
                "thumb_height": 420,
                "service_icon": "https:\/\/a.slack-edge.com\/e8ef6\/img\/unfurl_icons\/github.png",
                "id": 1
            }
        ],
        "ts": "1473174646.000011"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "You should see if\/when it starts\/stops along with each registered (admin\/non-admin) interface to the gRPC server.",
        "ts": "1473174713.000013"
    },
    {
        "type": "message",
        "user": "U11TAFPUP",
        "text": "Hi All,",
        "ts": "1473178986.000014"
    },
    {
        "type": "message",
        "user": "U11TAFPUP",
        "text": "The membersrvc.yml contains a lot of information. How do we protect this file?",
        "edited": {
            "user": "U11TAFPUP",
            "ts": "1473179712.000000"
        },
        "ts": "1473179031.000015"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "when I run \"make membersrvc\u201d command, I am getting the following error:\n\n# <http:\/\/github.com\/hyperledger\/fabric\/vendor\/github.com\/tecbot\/gorocksdb|github.com\/hyperledger\/fabric\/vendor\/github.com\/tecbot\/gorocksdb>\nvendor\/github.com\/tecbot\/gorocksdb\/options.go:328: not enough arguments in call to _Cfunc_rocksdb_options_set_compression_options\nmake: *** [build\/bin\/membersrvc] Error 2",
        "ts": "1473181325.000017"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "Any clue is highly appreciated.",
        "ts": "1473181342.000018"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "<@U1RLZ5729>: 2 things.\n1) Did the above info log worked? Or are you not there yet?",
        "ts": "1473181381.000019"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "2) Need some more info regarding the error your experience above... is this the latest code? etc... ?",
        "ts": "1473181421.000020"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "(which branch, etc.)",
        "ts": "1473181424.000021"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Are you running from vagrant, etc. Some more info as for how to 'reporoduce' this, might help...",
        "ts": "1473181459.000022"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "1. I was in between make thing. So could not make peers up. But will surely try and will let u know.\n2.  I am outside of vagrant. Running docker images from hub though.\nBut wanted to build from local fabric code. So was running \"make membersrvc\u201d to build membersrvc binaries. At that time, I got this error.",
        "ts": "1473181837.000023"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "The code is not just recent. A few days back, I had cloned it",
        "ts": "1473181866.000024"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Not a problem. Are you working on Linux\/Mac ?",
        "ts": "1473181966.000025"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "MAc",
        "ts": "1473181971.000026"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "What version of RocksDB have you installed (locally?)",
        "ts": "1473181997.000027"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "4.1",
        "ts": "1473182005.000028"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "I had searched this error online. They said earlier this function did not use one argument named \u201cmaxDictBytes\u201d and that needs to be added. But in my case, that argument is already there in my codes everywhere it is used. So I am wondering why that error is still coming?",
        "ts": "1473182237.000029"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "<https:\/\/github.com\/facebook\/rocksdb\/blob\/c3a4bea5dcf851b0236c682bf2fa4eba85a7c125\/db\/c.cc#L1575>",
        "attachments": [
            {
                "service_name": "GitHub",
                "title": "facebook\/rocksdb",
                "title_link": "https:\/\/github.com\/facebook\/rocksdb\/blob\/c3a4bea5dcf851b0236c682bf2fa4eba85a7c125\/db\/c.cc#L1575",
                "text": "rocksdb - A library that provides an embeddable, persistent key-value store for fast storage.",
                "fallback": "GitHub: facebook\/rocksdb",
                "thumb_url": "https:\/\/avatars2.githubusercontent.com\/u\/69631?v=3&s=400",
                "from_url": "https:\/\/github.com\/facebook\/rocksdb\/blob\/c3a4bea5dcf851b0236c682bf2fa4eba85a7c125\/db\/c.cc#L1575",
                "thumb_width": 400,
                "thumb_height": 400,
                "service_icon": "https:\/\/github.com\/apple-touch-icon.png",
                "id": 1
            }
        ],
        "ts": "1473182278.000030"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Yup, this is where they have added it. Are you sure you are replicating the exact environment with all the versions, etc. ? That might be a good check...",
        "ts": "1473182339.000032"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Our vendor\u2019ed version is from March 3rd and that new change was made on July 26. I think you have a different version installed locally.",
        "edited": {
            "user": "U11MANG9G",
            "ts": "1473182455.000000"
        },
        "ts": "1473182380.000033"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "<https:\/\/github.com\/hyperledger-archives\/fabric\/tree\/master\/vendor\/github.com\/tecbot\/gorocksdb>",
        "attachments": [
            {
                "service_name": "GitHub",
                "title": "hyperledger-archives\/fabric",
                "title_link": "https:\/\/github.com\/hyperledger-archives\/fabric\/tree\/master\/vendor\/github.com\/tecbot\/gorocksdb",
                "text": "THIS IS A READ-ONLY historic repository. Current development is at <https:\/\/gerrit.hyperledger.org\/r\/#\/admin\/projects\/fabric> . pull requests not accepted",
                "fallback": "GitHub: hyperledger-archives\/fabric",
                "thumb_url": "https:\/\/avatars3.githubusercontent.com\/u\/20729145?v=3&s=400",
                "from_url": "https:\/\/github.com\/hyperledger-archives\/fabric\/tree\/master\/vendor\/github.com\/tecbot\/gorocksdb",
                "thumb_width": 142,
                "thumb_height": 142,
                "service_icon": "https:\/\/github.com\/apple-touch-icon.png",
                "id": 1
            }
        ],
        "ts": "1473182472.000035"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Also, what\u2019s your go version?",
        "ts": "1473182566.000037"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "\u2026 etc., etc.  [You really need to make sure that you are replicating the exact same environment, dependencies, etc]",
        "ts": "1473182620.000038"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "No, that version is fine because I cloned this code sometym in August only. Moreover, I can see this change in code. That should not be the problem.",
        "ts": "1473182655.000039"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "My suspicion is that this is caused by some form of a version mismatch\u2026",
        "ts": "1473182724.000040"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "go version go1.7 darwin\/amd64",
        "ts": "1473182735.000041"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "I can\u2019t tell right away what does not match.",
        "ts": "1473182737.000042"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Here we go! :wink:",
        "ts": "1473182745.000043"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "what?",
        "ts": "1473182769.000044"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "You found it?",
        "ts": "1473182773.000045"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "????",
        "ts": "1473183077.000046"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "You found it. I don\u2019t think we use Go1.7 yet, for example.",
        "ts": "1473183278.000047"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "If you want to work `natively`, you should go over the Makefile and the dependencies and really match everything...",
        "ts": "1473183307.000048"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "E.g., <https:\/\/github.com\/hyperledger-archives\/fabric\/tree\/master\/scripts\/provision>",
        "attachments": [
            {
                "service_name": "GitHub",
                "title": "hyperledger-archives\/fabric",
                "title_link": "https:\/\/github.com\/hyperledger-archives\/fabric\/tree\/master\/scripts\/provision",
                "text": "THIS IS A READ-ONLY historic repository. Current development is at <https:\/\/gerrit.hyperledger.org\/r\/#\/admin\/projects\/fabric> . pull requests not accepted",
                "fallback": "GitHub: hyperledger-archives\/fabric",
                "thumb_url": "https:\/\/avatars3.githubusercontent.com\/u\/20729145?v=3&s=400",
                "from_url": "https:\/\/github.com\/hyperledger-archives\/fabric\/tree\/master\/scripts\/provision",
                "thumb_width": 142,
                "thumb_height": 142,
                "service_icon": "https:\/\/github.com\/apple-touch-icon.png",
                "id": 1
            }
        ],
        "ts": "1473183310.000049"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Along with the `vendor`ed packages, etc. It is a bit of a pain (I know!), which is why many prefer to work with the image \/ vagrant, etc.",
        "ts": "1473183356.000051"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "[Sorry I\u2019m doing several things at once. So might not respond to everything within seconds]",
        "ts": "1473183422.000052"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "Thats totally fine. Still it is of much help. So I should check all the versions and all now?",
        "edited": {
            "user": "U1RLZ5729",
            "ts": "1473183483.000000"
        },
        "ts": "1473183472.000053"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "Yes, that\u2019s what I\u2019d do. Start with the DB\/Go, etc\u2026 because I know of a change to the way pointers are handled (which relates to the error you see)",
        "ts": "1473183559.000055"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "It is not unheard of and we know 2 things: i. the code does build\/work\/run, etc\u2026 ii. Other people do work natively on Mac and were\/are able to continuously do so.",
        "ts": "1473183606.000056"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "<https:\/\/developer.ibm.com\/opentech\/2016\/06\/27\/running-hyperledger-fabric-natively-on-mac\/>",
        "attachments": [
            {
                "service_name": "IBM OpenTech",
                "title": "Running Hyperledger Fabric Natively on Mac - IBM OpenTech",
                "title_link": "https:\/\/developer.ibm.com\/opentech\/2016\/06\/27\/running-hyperledger-fabric-natively-on-mac\/",
                "text": "How to get Hyperledger Fabric running natively on a Mac using Docker for Mac (Beta)",
                "fallback": "IBM OpenTech: Running Hyperledger Fabric Natively on Mac - IBM OpenTech",
                "ts": 1467028640,
                "from_url": "https:\/\/developer.ibm.com\/opentech\/2016\/06\/27\/running-hyperledger-fabric-natively-on-mac\/",
                "service_icon": "https:\/\/developer.ibm.com\/opentech\/wp-content\/themes\/projectnext\/css\/favicon.ico",
                "id": 1
            }
        ],
        "ts": "1473183629.000057"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "[For example]",
        "ts": "1473183636.000059"
    },
    {
        "type": "message",
        "user": "U11MANG9G",
        "text": "So don\u2019t let these few errors [or my built-in single-task hardware\/brain] dissuade you\u2026 :wink:",
        "ts": "1473183696.000060"
    },
    {
        "type": "message",
        "user": "U1RLZ5729",
        "text": "ohh.. :smile:\nThanks for encouraging me.\nI would really try to fix this.",
        "ts": "1473183943.000061"
    },
    {
        "type": "message",
        "user": "U1LES78TT",
        "text": "can anyone here describe how the peers find the membersrvc to communicate with it? \nI'm having trouble with logging into a peer via CLI and REST\n    `rpc error: code = 2 desc = \"grpc: the client connection is closing\" `",
        "ts": "1473227745.000062"
    }
]